core/src/main/scala/scalaz/Injective.scala

```diff
@@ -14,32 +14,32 @@ case class Injective4[T[_,_,_,_]]()
-  implicit def Product1Injective   = Injective[Product1]()
-  implicit def Product2Injective   = Injective2[Product2]()
-  implicit def Product3Injective   = Injective3[Product3]()
-  implicit def Product4Injective   = Injective4[Product4]()
-  implicit def Product5Injective   = Injective5[Product5]()
-  implicit def SetInjective        = Injective[Set]()
-  implicit def Tuple2Injective     = Injective2[Tuple2]()
-  implicit def Tuple3Injective     = Injective3[Tuple3]()
-  implicit def Tuple4Injective     = Injective4[Tuple4]()
-  implicit def Tuple5Injective     = Injective5[Tuple5]()
-  implicit def ValidationInjective = Injective2[Validation]()
-  implicit def StreamInjective     = Injective[Stream]()
+  implicit def DisjunctionInjective: Injective2[\/] = Injective2[\/]()
+  implicit def EitherInjective: Injective2[Either] = Injective2[Either]()
+  implicit def FractionalInjective: Injective[Fractional] = Injective[Fractional]()
+  implicit def Function0Injective: Injective[Function0] = Injective[Function0]()
+  implicit def Function1Injective: Injective2[Function] = Injective2[Function1]()
+  implicit def Function2Injective: Injective3[Function2] = Injective3[Function2]()
+  implicit def Function3Injective: Injective4[Function3] = Injective4[Function3]()
+  implicit def Function4Injective: Injective5[Function4] = Injective5[Function4]()
+  implicit def IndexedSeqInjective: Injective[IndexedSeq] = Injective[IndexedSeq]()
+  implicit def IntegralInjective: Injective[Integral] = Injective[Integral]()
+  implicit def IterableInjective: Injective[Iterable] = Injective[Iterable]()
+  implicit def IteratorInjective: Injective[Iterator] = Injective[Iterator]()
+  implicit def ListInjective: Injective[List] = Injective[List]()
+  implicit def OptionInjective: Injective[Option] = Injective[Option]()
+  implicit def PartialFunctionInjective: Injective2[PartialFunction] = Injective2[PartialFunction]()
+  implicit def PartialOrderingInjective: Injective[PartialOrdering] = Injective[PartialOrdering]()
+  implicit def Product1Injective: Injective[Product1] = Injective[Product1]()
+  implicit def Product2Injective: Injective2[Product2] = Injective2[Product2]()
+  implicit def Product3Injective: Injective3[Product3] = Injective3[Product3]()
+  implicit def Product4Injective: Injective4[Product4] = Injective4[Product4]()
+  implicit def Product5Injective: Injective5[Product5] = Injective5[Product5]()
+  implicit def SetInjective: Injective[Set] = Injective[Set]()
+  implicit def Tuple2Injective: Injective2[Tuple2] = Injective2[Tuple2]()
+  implicit def Tuple3Injective: Injective3[Tuple3] = Injective3[Tuple3]()
+  implicit def Tuple4Injective: Injective4[Tuple4] = Injective4[Tuple4]()
+  implicit def Tuple5Injective: Injective5[Tuple5] = Injective5[Tuple5]()
+  implicit def ValidationInjective: Injective2[Validation] = Injective2[Validation]()
+  implicit def StreamInjective: Injective[Stream] = Injective[Stream]()
```

Comment:
We can remove the explicit tparams on the RHS; they should be inferred from the ascription.